meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -43,9 +71,56 @@ export const ShowStyleVariantsSettings = withTranslation()(
 			this.state = {
 				editedMappings: [],
 				timestampedFileKey: Date.now(),
+				dndVariants: this.getOrderedShowStyleVariants(),
 			}
 		}
 
+		componentDidUpdate(
+			prevProps: Readonly<Translated<IShowStyleVariantsProps>>,
+			prevState: Readonly<IShowStyleVariantsSettingsState>
+		) {
+			if (this.showStyleVariantsChanged(prevState, prevProps)) {
```

Comment:
After a refactor the check has been made into a separate method, this was to avoid setting the state too many times. But i can be done, so let me know if this is still preferred.